lib/SILOptimizer/PassManager/Passes.cpp

```diff
@@ -172,6 +172,13 @@ void AddHighLevelLoopOptPasses(SILPassManager &PM) {
   PM.addSwiftArrayOpts();
 }
 
+void AddFunctionSignaturePasses(SILPassManager &PM) {
+  // Make sure the run the rewriter to create the optimized functions before
+  // the cloner on the current function is run !.
+  PM.addFunctionSignatureOptRewriter();
+  PM.addFunctionSignatureOptCloner();
+}
```

Comment:
I believe the best place to place these (at least for initial experimentation) is:
- OptRewriter somewhere before the inliner in AddSSAPasses
- OptCloner pretty late in AddSSAPasses

This will allow us to rewrite to using the FSO-optimized function prior to attempting inlining, but allow us to fully optimized the current function before attempting to do FSO on the function.

What do you think?